bestofjs/bestofjs-webui
# Improve monthly graph on small screens

## Goal

Make the monthly graphs displayed in the project details page better on small screens.

Currently the numbers at the top and the months at the bottom are not displayed correctly.

![image](https://user-images.githubusercontent.com/5546996/185778305-7fd39669-92d5-4d58-bdb4-c3d50fb3a0eb.png)


## Changes

- The number at the top of the bar are no more displayed on small screens
- A tooltip is displayed when the users "hovers" the bar
- The  month and the year are displayed in a different way (see screenshot)

## How to test

Open the project detail page `/projects/:id` on a small screen and check that both graphs are displayed correctly:

- Stars added on GitHub, month by month
- NPM monthly downloads
 
A unit has been added to check the logic about the component that displays the year number at the bottom of the graph.

## Screenshots

### On small screens

![image](https://user-images.githubusercontent.com/5546996/185778400-2793db2b-aa4e-4a52-a634-1363f98d0b67.png)

![image](https://user-images.githubusercontent.com/5546996/185778407-c1cf3a3a-eb7a-42ba-a1e1-f74f27de677b.png)

### On bigger screens

![image](https://user-images.githubusercontent.com/5546996/185781195-d8141804-03ab-4140-b9f6-b85fbae72bab.png)

![image](https://user-images.githubusercontent.com/5546996/185781251-44dd2402-48d4-46d1-8247-db1d76041792.png)
